### ISSUE TITLE
Inconsistent window rule behavior

### ISSUE THREAD

**Issue by Mango0x45** [hyprwm/Hyprland#3036]
### Hyprland Version

Hyprland, built from branch  at commit 9654749244117f7f150c6f2a2ce4dede6e8cbb25 dirty (). Tag: v0.28.0  flags: (if any)

### Bug or Regression?

Bug

### Description

I have 3 window rules set with the aim of displaying a floating window in the middle of my screen.  Sometimes this behaves correctly, and sometimes it just… doesn’t.  Also a side, note but it’s noticeably ‘slow’ in that you can see the frame changing size.

This is what I have in my configuration:

    windowrule = float,        title=^(ec)$
    windowrule = size 50% 50%, title=^(ec)$
    windowrule = move 25% 25%, title=^(ec)$

This is what my expected result would be:

![2023-08-20_17:12:04](https://github.com/hyprwm/Hyprland/assets/57815710/af283d48-87e7-4a3a-be8d-c9502546effc)

This is what I get _sometimes_, but not always:

![2023-08-20_17:11:49](https://github.com/hyprwm/Hyprland/assets/57815710/34276f41-135f-471a-b719-ccf6eb3f45d8)

### How to reproduce

Just keep creating- and killing the window for 30 seconds or so.

**Comment by Mango0x45** [comment]
Ok so I am not unable to reproduce this after doing… nothing.  That being said, now I have the issue of the created window not actually being centred, but being off by a few pixels, with that pixel offset being seemingly random.  In other words, sometimes I get it in the middle, sometimes I get it a few pixels to the left of the middle, sometimes a few pixels to the right, etc.

**Comment by Mango0x45** [comment]
Interesting observation: I’ve enabled slide animations for my windows and now the window spawns in the center-top of my screen, half of it being cutoff. I am not sure if this is a coincidence or if it’s related.

**Comment by primalmotion** [comment]
I also have issue with windowrule beahving badly

```
windowrule = float,title:^(Picture-in-Picture)$
windowrule = pin,title:^(Picture-in-Picture)$
windowrule = size 640 360,title:^(Picture-in-Picture)$
windowrule = move 20 40,title:^(Picture-in-Picture)$
windowrule = noborder,title:^(Picture-in-Picture)$
windowrule = animation popin,title:^(Picture-in-Picture)$
```

This is what I use to manage FF PiP. It worked great. Now it partially work the first time (the window is floating, correctly positionned, but not correcly resized) Then if I close the PiP window and use it again, all these rules are not even run (the PiP window opened as a classic tiled window)

**Comment by MightyPlaza** [comment]
> I also have issue with windowrule beahving badly
> 
> ```
> windowrule = float,title:^(Picture-in-Picture)$
> windowrule = pin,title:^(Picture-in-Picture)$
> windowrule = size 640 360,title:^(Picture-in-Picture)$
> windowrule = move 20 40,title:^(Picture-in-Picture)$
> windowrule = noborder,title:^(Picture-in-Picture)$
> windowrule = animation popin,title:^(Picture-in-Picture)$
> ```
> 
> This is what I use to manage FF PiP. It worked great. Now it partially work the first time (the window is floating, correctly positionned, but not correcly resized) Then if I close the PiP window and use it again, all these rules are not even run (the PiP window opened as a classic tiled window)

totally unrelated and firefox issue https://github.com/hyprwm/Hyprland/issues/2942#issuecomment-1677625225

**Comment by primalmotion** [comment]
well... it does not happen with the latest tagged release, and I don't see the initialTitle set to firefox with my version. It is correctly set to "Picture-in-Picture", so that does not look like #2942

**Comment by MightyPlaza** [comment]
are you sure?
for me since, I think, 3 firefox releases ago on arch, initialTitle is different when launching PiP for the 1st and 2nd times

**Comment by primalmotion** [comment]
Ok I dug a bit more. I use ff from flatpak, and I do eventually have the issue of the initialTitle. BUT, when I'm not, my rules are still not applied correctly. 

I said:

> Now it partially work the first time (the window is floating, correctly positionned, but not correcly resized) Then if I close the PiP window and use it again, all these rules are not even run (the PiP window opened as a classic tiled window)

So the last part is indeed due to the bug you mentionned, but the first part seems to be just bad window rule behavior (it's not resized)

**Comment by MightyPlaza** [comment]
the rules seem to work for me (1st time)

**Comment by KrulYuno** [comment]
I've had this kind of issue ever since I used hyprland to test it. I thought it's normal and didn't bother. Mine just happens the longer you use an app, in this case for me it's kitty terminal, and the issue just comes in.

**Comment by fleurc** [comment]
Idk if this applies, but it also involves Window Rules.

When opening a web-app from Vivaldi, such as Youtube Music, it ends up opening the window in floating. I've tried fixing it with `windowrulev2=pseudo, class:^(vivaldi-stable)$` and similar variations, tried with `tile` also but to no avail.

after giving up and erasing the rules from my config, I learned that opening wf-recorder to record the issue makes the window go back to tiling. So I have the recording from my phone of both occurrences.

https://github.com/hyprwm/Hyprland/assets/35513885/3da29fa4-0ade-419c-94f2-990fadef084a
